File: IntToString/IntToStringv1.java
public class IntToStringv1 {
    public static void main(String[] args){
        System.out.println(numbersToWords(123));
        System.out.println(numbersToWords(12345));
        System.out.println(numbersToWords(1234567));
    }

    public static String numbersToWords(int num){
        String[] spelledNumbers = {"zero",
            "one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten",
            "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen", "eighteen", "nineteen", "twenty",
            "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety",
            "one hundred", "one thousand"
        };

        if (num >= 0 && num < 21) {return spelledNumbers[num];}
        else if (num < 100) {
            return spelledNumbers[20 + num / 10] + (num % 10 == 0 ? "" : " " + spelledNumbers[num % 10]);
        }
        else if (num < 1000) {
            return spelledNumbers[num / 100] + " hundred" + (num % 100 == 0 ? "" : " and " + numbersToWords(num % 100));
        }
        else if (num < 10000) {
            return spelledNumbers[num / 1000] + " thousand" + (num % 1000 == 0 ? "" : " " + numbersToWords(num % 1000));
        }
        else {return "Number out of range";}
    }
}

File: 273. IntToString/IntToStringv1.py
units = {"", "One", "Two", "Three", "Four", "Five", "Six", "Seven", "Eight", "Nine"}
teens = {"Ten", "Eleven", "Twelve", "Thirteen", "Fourteen", "Fifteen", "Sixteen", "Seventeen", "Eighteen", "Nineteen"}
tens = {"", "Ten", "Twenty", "Thirty", "Forty", "Fifty", "Sixty", "Seventy", "Eighty", "Ninety"}

def IntToString(num):
    if num == 0: return "Zero"
    result = ""
    if num < 0: 
        result += "Minus "
        num*= (-1)
    StringCreator(num, result)
    return result.strip()

def StringCreator(num, result):
    if num >= 1000000000:
        StringCreator(num // 1000000000, result)
        result += "Billion "
        num % 1000000000
    elif num >= 1000000:
        StringCreator(num // 1000000, result)
        result += "Million "
        num % 1000000
    elif num >= 1000:
        StringCreator(num // 1000, result)
        result += "Thousand "
        num % 1000
    elif num >= 100:
        StringCreator(num // 100, result)
        result += str(units[num // 100]) + "Hundred "
        num % 100
    elif num >= 20:
        StringCreator(num // 100, result)
        result += str(tens[num // 10]) +  " "
        num % 10
        num % 100
    elif num >= 10:
        StringCreator(num // 100, result)
        result += str(teens[num // 10]) +  " "
        num % 10
    else:
        StringCreator(num // 100, result)
        result += " "

n = int(input("Put a number: "))
print(f"{n} as text is: {IntToString(n)}")
